### ISSUE TITLE
Make External Legend Container Fit Content

### ISSUE THREAD

**Issue by bfailing** [amcharts/amcharts4#1202]
See [CodePen here](https://codepen.io/anon/pen/gyGWzE).  I'm trying to make the external legend container fit the content which may dynamically change in width (and potentially height).  Is there a way to make the container (shown with a gray background) the same size as the content (i.e. the area that causes the cursor to change to a pointer when hovering)?

**Comment by martynasma** [comment]
Take a look at this tutorial which deals with automatic sizing of external legend container:

https://www.amcharts.com/docs/v4/tutorials/chart-legend-in-an-external-container/

**Comment by bfailing** [comment]
I did.  That's where I got the code.

I enabled the two remaining portions in [this pen](https://codepen.io/anon/pen/ZZXRrO) so that it should be the same as the tutorial.

A few issues remain:

1. This doesn't account for the width.  Again, it would be nice to have the container (not the div that we create, but whatever elements your code adds automatically) be sized to just the portion that causes the cursor to change (and is clickable).  It looks like the height and width of that rect element are specified using inline attributes, so wouldn't it be possible to limit the parent elements (created automatically by your code) to just those dimensions?

2. It starts off with a height of 0.  If the browser window is resized, it seems to get the correct height.  Presumably because it sized it before the content loaded.  However, even if this did work, it's a band-aid fix.  It would be better to just get rid of (or disable) whatever is causing the unnecessary height in your code.  Else, is there a way to run the resizeLegend function later after the content is loaded?
